urllib3/util/selectors.py

```diff
@@ -534,20 +551,73 @@ def close(self):
             self._kqueue.close()
             super(KqueueSelector, self).close()
 
+    KqueueSelector = _KqueueSelector
+
+
+def _detect_select_patch(*args, **kwargs):
+    # If monkey-patching occurs then the canary won't be
+    # defined within the new select module. Because the
+    # module was patched we now have to re-evaluate which
+    # selectors are available.
```

Comment:
I guess the alternative is determining `DefaultSelector` on each call as you mentioned in the issue.